spring-cloud-security/src/main/java/org/springframework/cloud/security/oauth2/client/ResourceServerTokenRelayAutoConfiguration.java

```diff
@@ -65,20 +69,36 @@
 @ConditionalOnWebApplication
 public class ResourceServerTokenRelayAutoConfiguration {
 
-	@Bean
+	protected static final String TOKEN_RELAY_REQUEST_INTERCEPTOR = "tokenRelayRequestInterceptor";
+
+	@Bean(name = TOKEN_RELAY_REQUEST_INTERCEPTOR)
 	public HandlerInterceptor tokenRelayRequestInterceptor(
 			final OAuth2ClientContext context) {
 		final AccessTokenContextRelay relay = new AccessTokenContextRelay(context);
 		return new HandlerInterceptorAdapter() {
 			@Override
 			public boolean preHandle(HttpServletRequest request,
-					HttpServletResponse response, Object handler) throws Exception {
+									 HttpServletResponse response, Object handler) throws Exception {
 				relay.copyToken();
 				return true;
 			}
 		};
 	}
 
+	@Configuration
+	public static class ResourceServerTokenRelayRegistrationAutoConfiguration extends WebMvcConfigurerAdapter {
+
+		@Autowired
```

Comment:
Yes, I agree in my first version I removed the bean. I only added it back because in my unit test https://github.com/spring-cloud/spring-cloud-security/pull/124/files#diff-65eaf7264f6654f241387c2800cc20b0R70 I want to check with a spy that it is really this bean that is called. I experimented if there is any way to access the InterceptorRegistry from the unit test so I could get the filter from there, but I haven't found the right way.